classes/order/OrderInvoice.php

```diff
@@ -140,14 +140,14 @@ public function getProductsDetail()
         $id_lang = Context::getContext()->language->id;
 
         return Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS('
-		SELECT *, pl.`name` as product_name
+		SELECT *
 		FROM `'._DB_PREFIX_.'order_detail` od
 		LEFT JOIN `'._DB_PREFIX_.'product` p
 		ON p.id_product = od.product_id
 		LEFT JOIN `'._DB_PREFIX_.'product_shop` ps ON (ps.id_product = p.id_product AND ps.id_shop = od.id_shop)
 		LEFT JOIN `'._DB_PREFIX_.'product_lang` pl ON (pl.id_product = p.id_product AND pl.id_lang = '.$id_lang.' AND pl.id_shop = od.id_shop)
```

Comment:
And do you know why this is totally correct?
Because this clause cannot change anything. The product name's language doesn't depend on the SQL clause, **only** on the shop language chosen in frontend. If the customer chooses English for instance, the product will be saved with the full English name in the database table  **ps_order_detail** as __product-name__. It is that simple! I should have known better! :(